tests/sarracenia/config_test.py

```diff
@@ -90,6 +91,265 @@ def test_variableExpansion():
+
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "declare env VAR99=hoho" )
+     assert( options.env['VAR99'] == 'hoho' )
+
+def test_read_line_flags():
+
+     options = sarracenia.config.default_config()
+
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "download off" )
+     assert( options.download == False )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "download on" )
+     assert( options.download == True )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "download False" )
+     assert( options.download == False )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "download True" )
+     assert( options.download == True )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "download no" )
+     assert( options.download == False )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "download yes" )
+     assert( options.download == True )
+
+     assert( options.acceptSizeWrong == False )
+     assert( options.acceptUnmatched == True )
+     assert( options.sourceFromExchange == False )
+     assert( options.sourceFromMessage == False )
+     assert( options.sundew_compat_regex_first_match_is_zero == False )
+     assert( options.topicCopy == False )
+
+def test_read_line_counts():
+
+     options = sarracenia.config.default_config()
+
+     # crasher input:
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch -1" )
+     assert( options.batch == -1 )
+
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 1" )
+     assert( options.batch == 1 )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 1kb" )
+     assert( options.batch == 1024 )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 1k" )
+     assert( options.batch == 1000 )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 1m" )
+     assert( options.batch == 1000000 )
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 1mb" )
+     assert( options.batch == 1024*1024 )
+
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 134684" )
+     assert( options.batch == 134684 )
+
+     # count truncates floats... is this a good thing?
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "batch 1.9" )
+     assert( options.batch == 1 )
+
+
+def test_read_line_sets():
+
+     options = sarracenia.config.default_config()
+     logger.info( f" {options.fileEvents=} " )
+
+     assert( options.fileEvents == set( ['create', 'delete', 'link', 'mkdir', 'modify', 'rmdir' ] ) )
+
+     options.parse_line( "subscribe", "ex1", "subscribe/ex1", 1, "fileEvents -rmdir" )
+     logger.info( f" {options.fileEvents=} " )
+     assert( options.fileEvents == set( ['create', 'delete', 'link', 'mkdir', 'modify' ] ) )
+
+     #FIXME: must print an error message... how to test for that?
```

Comment:
You can capture logs using the `caplog` fixture. I've used it in the `retry` driver tests, like [here](https://github.com/MetPX/sarracenia/blob/4dea8827765ab1b7ac05a8705f20e07be5cf8765/tests/sarracenia/diskqueue_test.py#L236), [here](https://github.com/MetPX/sarracenia/blob/4dea8827765ab1b7ac05a8705f20e07be5cf8765/tests/sarracenia/diskqueue_test.py#L260), and [here](https://github.com/MetPX/sarracenia/blob/4dea8827765ab1b7ac05a8705f20e07be5cf8765/tests/sarracenia/flowcb/retry_test.py#L313).

You set a variable to say if it's there (`False` by default), and then check if a particular message exists in the logs, setting the variable to `True` if it does, and then finally asserting on that variable.

Reply:
will do that as action later... change is still good.